doc/developer-guide/api/functions/TSHttpTxnTypeGet.en.rst

```diff
@@ -0,0 +1,37 @@
+.. Licensed to the Apache Software Foundation (ASF) under one or more
+   contributor license agreements.  See the NOTICE file distributed
+   with this work for additional information regarding copyright
+   ownership.  The ASF licenses this file to you under the Apache
+   License, Version 2.0 (the "License"); you may not use this file
+   except in compliance with the License.  You may obtain a copy of
+   the License at
+
+   http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+   implied.  See the License for the specific language governing
+   permissions and limitations under the License.
+
+.. include:: ../../../common.defs
+
+.. default-domain:: c
+
+TSHttpTxnTypeGet
```

Comment:
I pointed this out on the dev list but I'm commenting here as well for visibility.

I don't think transactions on a connection/session don't have different types. Shouldn't it a type of a connection, a session, or a port?

Reply:
Hmm.  Guess I called it Txn, since I wanted to call this from a hook where I have a txn object.  I want to distinguish between tunnel type (which is a connection characteristic) and a HTTP transaction (which I guess is a characteristic of the connection as well). However from an implementation perspective this is information we pull from the HttpSM.  Let me think a bit on how invasive it would be to add this information to the vconn.

Reply:
Added a test case for the CONNECT method.  This gets processed as a HTTP Request type even though it does eventually set up a tunnel.  The tr-pass case and sni config case will get reported as various tunnel transactions since they are processed as tunnels directly not a transaction sent to the proxy first.

Reply:
Just to confirm, does that mean `TS_TXN_TYPE_HTTP` will be returned on a transaction that receives CONNECT method?

I'm wondering if that is the expected and desired behavior. In other words, would you mind if somebody added `TS_TXN_TYPE_CONNECT_TUNNEL`?

In either case, it would be nice to have the behavior documented.

Reply:
Correct, as it stands a txn created from a CONNECT request will be reported as TS_TXN_TYPE_HTTP.  If one wanted to look at the method, they could differentiate that as TS_TXN_TYPE_CONNECT_TUNNEL.  I would be ok with that addition.  In the meantime, I'll review my proposed documentation to make sure this point is clear.